sdk/storage/Azure.Storage.Blobs/tests/ServiceClientTests.cs

```diff
@@ -67,19 +67,13 @@ public async Task ListContainersSegmentAsync_Marker()
             using (this.GetNewContainer(out var container, service: service))
             {
                 var marker = default(string);
-                ContainersSegment containersSegment;
-
                 var containers = new List<ContainerItem>();
 
-                do
+                var enumerator = service.GetContainersAsync().ByPage(marker).GetAsyncEnumerator();
```

Comment:
Not a bad thing to test for, but why not just use the normal await foreach?

Reply:
No good reason - I just wanted to try one manually.  I'll change it back to an await foreach.